controllers/module_reconciler.go

```diff
@@ -82,10 +85,8 @@ func (r *ModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	logger := log.FromContext(ctx)
 
-	mod := ootov1alpha1.Module{}
-
-	if err := r.Client.Get(ctx, req.NamespacedName, &mod); err != nil {
-		logger.Error(err, "Could not get module")
+	mod, err := r.getRequestedModule(ctx, req.NamespacedName)
+	if err != nil {
 		return res, err
```

Comment:
We should probably wrap the error here.

Reply:
not sure, it is wrapped and pretty much explained in the called function, we are not going to get any new info from wrapping it

Reply:
IMO this is not just for "adding info" but also to make it easier to track the error in the code base and understand the exact failure point instead of guessing.

Reply:
fixed